src/indra_cogex/sources/disgenet/__init__.py

```diff
@@ -0,0 +1,186 @@
+    "VD_variant2GeneIdSet": int,
+    "VD_commonVariantIdSet": int,
+    "VD_jaccard": float,
+    "VD_pvalue": float,
+}
+
+
+class DisgenetProcessor(Processor):
+    """Processor for the DisGeNet database."""
+
+    name = "disgenet"
+    node_types = ["BioEntity"]
+    gene_relation = "gene_disease_association"
+    variant_relation = "variant_disease_association"
+
+    def __init__(self):
+        """Initialize the DisGeNet processor."""
+        self.gene_df = load_disgenet_disease_gene(
+            CURATED_DISEASE_GENES_ASSOCIATIONS_URL
+        )
+
+    def get_nodes(self):  # noqa:D102
+        diseases = {
+            tuple(row)
+            for row in self.gene_df[
+                ["disease_prefix", "disease_id", "disease_name"]
+            ].values
+        }
+        for prefix, identifier, name in diseases:
+            yield Node.standardized(
+                db_ns=prefix, db_id=identifier, name=name, labels=["BioEntity"]
+            )
+        for hgnc_id in self.gene_df["hgnc_id"].unique():
+            yield Node.standardized(db_ns="HGNC", db_id=hgnc_id, labels=["BioEntity"])
+
+    def get_relations(self):  # noqa:D102
+        columns = [
+            "hgnc_id",
+            "disease_prefix",
+            "disease_id",
+            "NofSnps",
+            "NofPmids",
+        ]
+        for hgnc_id, disease_prefix, disease_id, snps, papers in self.gene_df[
+            columns
+        ].values:
+            data = {"snps:int": snps, "source": self.name, "papers:int": papers}
+            yield Relation(
+                "HGNC", hgnc_id, disease_prefix, disease_id, self.gene_relation, data
+            )
+
+
+class DisgenetVariantProcessor(Processor):
+    """Processor for the DisGeNet database."""
+
+    name = "disgenet_variant"
+    variant_df: pd.DataFrame
+    node_types = ["BioEntity"]
+    relation = "variant_disease_association"
+
+    def __init__(self):
+        """Initialize the DisGeNet processor."""
+        self.variant_df = load_disgenet_disease_gene(
+            CURATED_DISEASE_VARIANT_ASSOCIATIONS_URL, variant=True
+        )
+
+    def get_nodes(self):  # noqa:D102
+        # Since there exists no column in the disease-variant association tsv
+        # that describes disease prefix, opted to use diseaseClass
+        diseases = {
+            tuple(row)
+            for row in self.variant_df[
+                ["diseaseClass", "diseaseId", "diseaseName"]
+            ].values
+        }
+        for disease_class, identifier, name in diseases:
+            yield Node.standardized(
+                db_ns=disease_class, db_id=identifier, name=name, labels=["BioEntity"]
+            )
+        for snp_id in self.variant_df["snpId"].unique():
+            yield Node.standardized(
+                db_ns="DBSNP", db_id=snp_id, name=snp_id, labels=["BioEntity"]
+            )
+
+    def get_relations(self):  # noqa:D102
+        # Use diseaseName instead of diseasePrefix as diseasePrefix does not
```

Comment:
the point of the UMLS mapper was to add standardized fields `disease_prefix`, `disease_id`, and `disease_name`, which aren't the same as the camel-cased ones

a trick for when stuff isn't what you expect is to print(df.head()) to get a look at some sample data